packages/components/src/table/style.scss

```diff
@@ -6,96 +6,58 @@
 		position: relative;
 	}
 
+	.woocommerce-card__header {
+		padding: ($gap / 2) $gap;
+	}
+
 	.woocommerce-search {
 		flex-grow: 1;
 	}
 
 	.woocommerce-card__action {
-		justify-self: flex-end;
-		margin: -($gap - 3) 0;
-	}
-
-	.woocommerce-card__menu {
-		justify-self: flex-end;
+		margin-left: auto;
```

Comment:
This seems to remove the space between the table title and the `Compare` button.

With:
![image](https://user-images.githubusercontent.com/3616980/53302141-6d649b00-385b-11e9-8c59-40a099cbb9c2.png)

Without:
![image](https://user-images.githubusercontent.com/3616980/53302136-62116f80-385b-11e9-83ec-6c5a4b55cd75.png)

Reply:
Good eyes!  👀 

Updated in https://github.com/woocommerce/wc-admin/pull/1653/commits/f2f4a8e8d754f8d8035d8f5947c81acff530f006